### DISCUSSION TITLE
Send json object

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5714] [Q&A] [answered]
I come for help
I have 1 channel destined to send access credentials for authentication, these must go in the request body and in JSON format and I want them to be sent every 5 minutes.
The query is:
What would be the most optimal way to configure the channel, what send this message?
Can you create a javascript reader type connector and program what would be the JSON object and the execution of the channel itself? Or is there another way?

**Comment by jonbartels** [comment, marked as answer]
There are a few ways to do this:

1. JavaScript Reader on a poll. You can have your JS reader just do `return "wooo any string"` with a RAW message datatype then an HTTP destination can renew your token
2. A better way is that when you get a token, you should cache that token in the globalChannelMap or globalMap. Then also cache a timestamp for when the token expires. Then you can have a filter that checks if the token exists and if the expiration time is either soon or now. Then if the renewal time is close, you can call a destination to renew the token before you call the destination to actually send the data. 

Both approaches have the same result. The difference is that one runs on a poller, the other one just checks when you try to use the token.

**Reply by pacmano1** [reply]
That assumes the expiration time is known, some systems don't provide that.   Also some systems extend the life of the token every time you use it.    It also assumes the token, although not expired, still works.   I would have a token poller on a fixed time basis.   On its next poll it skips the token get but calls some other API endpoint, checking that the call was successful, and if not due to token expiration on next poll have it get a fresh token (or force another poll for a fresh token).  

Also you can use a JS reader of data type JSON and just "return true".   It does not need to be raw.
